src/pepr/policies/security.ts

```diff
@@ -25,7 +25,24 @@ import { exemptionAnnotationPrefix, isExempt, markExemption } from "./exemptions
  */
 When(a.Pod)
   .IsCreatedOrUpdated()
-  .Mutate(markExemption(Policy.DisallowPrivileged))
+  .Mutate(request => {
+    markExemption(Policy.DisallowPrivileged)(request);
+    if (request.HasAnnotation(`${exemptionAnnotationPrefix}.${Policy.DisallowPrivileged}`)) {
+      return;
+    }
+
+    // Check if any containers defined in the pod do not have the `allowPrivilegeEscalation` field present. If not, include it and set to false.
+    for (const container of containers(request)) {
+      container.securityContext = container.securityContext || {};
+      if (
+        container.securityContext.allowPrivilegeEscalation === undefined &&
+        !container.securityContext.privileged
+      ) {
+        container.securityContext.allowPrivilegeEscalation = false;
+      }
+    }
+    annotateMutation(request, Policy.DisallowPrivileged);
+  })
   .Validate(request => {
```

Comment:
We typically still validate a given condition, even if our mutation should guarantee it (see [drop all capabilities](https://github.com/defenseunicorns/uds-core/blob/fix/527-ctx-priv-esc/src/pepr/policies/security.ts#L330-L367) for example - we both mutate and validate that all capabilities are dropped).

I think we would want to check for this undefined case and deny workloads that have it undefined in our Validate (in addition to the mutation you already have). This keeps us from just assuming the mutation worked and ensures that our validate is strictly blocking the scenario, rather than assuming well-formed pods are already coming through it.

Reply:
I see what you mean. I'll update `violations` to be more specific.